lib/rules/space-infix-ops.js

```diff
@@ -8,6 +8,23 @@
 // Rule Definition
 //------------------------------------------------------------------------------
 
+const OPERATORS = [
+    "*", "/", "%", "+", "-", "<<", ">>", ">>>", "<", "<=", ">", ">=",
+    "==", "!=", "===", "!==", "&", "^", "|", "&&", "||", "=",
+    "+=", "-=", "*=", "/=", "%=", "<<=", ">>=", ">>>=", "&=", "^=", "|=",
+    "?", ":", ",", "**"
+];
+
+const operatorEnum = { enum: ["always", "never", "ignore"] };
+const schemaProperties = {
+    int32Hint: { type: "boolean" },
+    all: operatorEnum
+};
+
+OPERATORS.forEach(operator => {
+    schemaProperties[operator] = operatorEnum;
```

Comment:
You are correct that my rule uses option 2.
I added a test and verified that this works.